src/middlewares/auth.middleware.js

```diff
@@ -0,0 +1,52 @@
+import ResponseService from '../services/response.service';
+import UserService from '../services/user.service';
+import JwtService from '../services/jwt.service';
+
+const loginMiddleware = {
+  userExist: async (req, res, next) => {
+    const user = await UserService.findByEmail(req.body.email.trim());
+    if (!user) {
+      ResponseService.setError(401, 'You don\'t have an account. Please create an account');
+      return ResponseService.send(res);
+    }
+    next();
+  },
+  isAccountActive: async (req, res, next) => {
+    const user = await UserService.findByEmail(req.body.email.trim());
+    if (!user.isVerified) {
+      ResponseService.setError(400, 'You have not activated your account. Please activate your account first');
+      return ResponseService.send(res);
+    }
+    next();
+  },
+
+  isloggedIn: async (req, res, next) => {
+    let token = req.headers.authorization || req.headers['x-access-token'] || req.body.token || req.query.slt;
```

Comment:
Why are there multiple entry points for setting an authorization token? There should only be one entry point and it must be set as a header.

Reply:
Ok. this also solve the other comment in test.